Clarifies events page

akash19coder opened the pull request:
<!-- If your PR fixes an open issue, use `Closes #999` to link your PR with the issue. #999 stands for the issue number you are fixing -->

## Fixes Issue  

closes #5393 

<!-- Remove this section if not applicable -->

<!-- Example: Closes #31 -->

## Changes proposed
 Added clarity to the events page by introducing a dropdown.
Create a DropDown component.

<!-- List all the proposed changes in your PR -->

<!-- Mark all the applicable boxes. To mark the box as done follow the following conventions -->
<!--
[x] - Correct; marked as done
[X] - Correct; marked as done

[ ] - Not correct; marked as **not** done
-->

## Check List (Check all the applicable boxes) <!-- Follow the above conventions to check the box -->

- [X] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] All new and existing tests passed.
- [X] This PR does not contain plagiarized content.
- [X] The title of my pull request is a short description of the requested changes.

## Screenshots

<!-- Add all the screenshots which support your changes -->

## Note to reviewers

<!-- Add notes to reviewers if applicable -->


<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/7180"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

amandamartin-dev reviewed (dismissed):
A few notes.  Would it be possible to only have the dropdown show valid choices?  It's not a great UX to click through many that have "no events found".

Another thing I noticed was css related.  Some of the text in the dropdown overlaps the carrot (I'll add some screenshots).  There could also be some more space between the colon and dropdown.

And lastly just a point for discussion - this looks a little unusual to me on mobile although I'm not sure a better solution off the top of my head. Would love another set of eyes here wdyt @eddiejaoude 

mobile view:
![mobile](https://github.com/EddieHubCommunity/LinkFree/assets/97615019/d7526c17-f547-4fb1-ac94-9b85e6163b38)

style overlap/spacing:
![styling issues](https://github.com/EddieHubCommunity/LinkFree/assets/97615019/a018471f-7424-4181-9966-ef5ed084f615)

akash19coder commented:
> A few notes. Would it be possible to only have the dropdown show valid choices? It's not a great UX to click through many that have "no events found".

I have changed the dropdown to show only the valid choices.

> Another thing I noticed was css related. Some of the text in the dropdown overlaps the carrot (I'll add some screenshots). There could also be some more space between the colon and dropdown.

Fixed this also. Added more space between colon and dropdown.

eddiejaoude reviewed (approved):
Looks great 👍  thank you

eddiejaoude commented:
> And lastly just a point for discussion - this looks a little unusual to me on mobile although I'm not sure a better solution off the top of my head. Would love another set of eyes here wdyt @eddiejaoude

Thanks @amandamartin-dev for reviewing! Yep the double dropdown doesn't look quite right, but I am not sure of another solution - I don't think it is a block for now, we can get it out and people can feedback with ideas (we are also looking into getting a redesign of the app at the moment so it could get moped up with that)

eddiejaoude commented:
Changes look good...


<img width="454" alt="Screenshot 2023-05-30 at 10 30 22" src="https://github.com/EddieHubCommunity/LinkFree/assets/624760/e9f383d0-3f40-49cc-9fc8-0cd119f8e02e">
<img width="1017" alt="Screenshot 2023-05-30 at 10 30 10" src="https://github.com/EddieHubCommunity/LinkFree/assets/624760/bdf88912-cef5-4abc-b0aa-574c9f4a197f">

eddiejaoude commented:
@akash19coder please can you resolve the conflict. I think it acme from this PR https://github.com/EddieHubCommunity/LinkFree/pull/7135

akash19coder commented:
@eddiejaoude I have resolved all the conflicts
😄

dan-mba commented on components/user/UserEvents.js:
```diff
@@ -1,38 +1,92 @@
+    setEventType(event.target.value);
+  };
+
+  const filterByEventType = (event, eventType) => {
```
Doing Date math inside a server-side rendered component is a bad idea.

Please change this section to look like:
```js
  const filterByEventType = (event, eventType) => {
    switch (eventType) {
      case "future":
        return event.date.future;
      case "ongoing":
        return event.date.ongoing;
      case "virtual":
        return event.date.future && event.isVirtual;
      case "inPerson":
        return event.date.future && event.isInPerson;
      case "cfpOpen":
        return event.date.cfpOpen;
      case "free":
        return event.price?.startingFrom === 0;
      case "paid":
        return event.price?.startingFrom > 0;
      case "past":
        return !event.date.future;
      default:
        return true;
    }
  };
```

akash19coder replied:
@dan-mba I have changed the function to lool like what you suggested

dan-mba reviewed (approved):
Looks good to me
